Sources/PT.PM.Common/UstConstantFolder.cs

```diff
@@ -234,14 +240,26 @@ private FoldResult TryFoldUnaryOperatorExpression(UnaryOperatorExpression unaryO
 
             if (unaryOperatorExpression.Operator.UnaryOperator == UnaryOperator.Minus)
             {
-                if (foldResult.Value is long longValue)
+                object valueToProcess = null;
+                switch (foldResult.Value)
                 {
-                    return ProcessUnaryExpression(unaryOperatorExpression, foldResult, -longValue);
+                    case double doubleVal:
+                        valueToProcess = -doubleVal;
```

Comment:
Excess boxing (conversion to object). It's better to replace with return statements.